lib/langchain/llm/openai.rb

```diff
@@ -181,8 +180,11 @@ def compose_parameters(model, params, &block)
       parameters = default_params.merge(params)
 
       if block
+        @response_chunks = []
         parameters[:stream] = proc do |chunk, _bytesize|
-          yield chunk.dig("choices", 0)
+          chunk_content = chunk.dig("choices", 0)
+          @response_chunks << chunk
```

Comment:
It looks like in order to accomodate for `n > 1`, we'd need reconstruct the chunks based on their index:
```ruby
irb(main):007:1* response = llm.chat(prompt: "hello! How are you?", n: 2, temperature: 1)  do |chunk|
irb(main):008:1*   puts chunk
irb(main):009:0> end
{"index"=>0, "delta"=>{"role"=>"assistant", "content"=>""}, "finish_reason"=>nil}
{"index"=>0, "delta"=>{"content"=>"Hello"}, "finish_reason"=>nil}
{"index"=>1, "delta"=>{"role"=>"assistant", "content"=>""}, "finish_reason"=>nil}
{"index"=>1, "delta"=>{"content"=>"Hello"}, "finish_reason"=>nil}
{"index"=>0, "delta"=>{"content"=>"!"}, "finish_reason"=>nil}
{"index"=>1, "delta"=>{"content"=>"!"}, "finish_reason"=>nil}
{"index"=>0, "delta"=>{"content"=>" I"}, "finish_reason"=>nil}
...
```